### ISSUE TITLE
Supybot Backend

### ISSUE THREAD

**Issue by Nolski** [chaoss/grimoirelab-perceval#709]
# Overview

When running `perceval supybot` I receive a parse error saying a date is expected on line one.

## Further details

The exact output of the command is below:

```bash
$ perceval supybot ritfoss ./supybot/logs/ChannelLogger/freenode/#rit-foss/
[2021-01-12 11:58:22,013] - Sir Perceval is on his quest.
[2021-01-12 11:58:22,016] - Fetching messages of 'ritfoss' from 1970-01-01 00:00:00+00:00
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/perceval/backends/core/supybot.py", line 207, in parse_supybot_log
    for message in parser.parse():
  File "/usr/local/lib/python3.7/dist-packages/perceval/backends/core/supybot.py", line 366, in parse
    ts, msg = self._parse_supybot_timestamp(line)
  File "/usr/local/lib/python3.7/dist-packages/perceval/backends/core/supybot.py", line 387, in _parse_supybot_timestamp
    raise ParseError(cause=msg)
perceval.errors.ParseError: date expected on line 1

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/perceval/backend.py", line 618, in run
    for item in big.items:
  File "/usr/local/lib/python3.7/dist-packages/perceval/backend.py", line 794, in __fetch
    raise e
  File "/usr/local/lib/python3.7/dist-packages/perceval/backend.py", line 788, in __fetch
    for item in items:
  File "/usr/local/lib/python3.7/dist-packages/perceval/backend.py", line 226, in fetch
    for item in self.fetch_items(category, **kwargs):
  File "/usr/local/lib/python3.7/dist-packages/perceval/backends/core/supybot.py", line 118, in fetch_items
    for message in self.parse_supybot_log(archive):
  File "/usr/local/lib/python3.7/dist-packages/perceval/backends/core/supybot.py", line 211, in parse_supybot_log
    raise ParseError(cause=cause)
perceval.errors.ParseError: file: ./supybot/logs/ChannelLogger/freenode/#rit-foss/#rit-foss.log; reason: date expected on line 1

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/bin/perceval", line 169, in <module>
    main()
  File "/usr/local/bin/perceval", line 115, in main
    cmd.run()
  File "/usr/local/lib/python3.7/dist-packages/perceval/backend.py", line 630, in run
    raise RuntimeError(str(e))
RuntimeError: file: ./supybot/logs/ChannelLogger/freenode/#rit-foss/#rit-foss.log; reason: date expected on line 1
```

However, the log file seems to be formatted correctly with a datetime at the beginning of each line as shown below:

```bash
$ cat ./supybot/logs/ChannelLogger/freenode/#rit-foss/#rit-foss.log
2021-01-12T11:28:19  *** grimoire_bot <grimoire_bot!~limnoria@45.55.49.14> has joined #rit-foss
2021-01-12T11:29:41  <nolski> test
2021-01-12T11:29:41  <xbot1313> Test Case Passed!
```

**Comment by gregoriorobles** [comment]
I've had a look at the tests, and I see that the problem here is that the fact that the timezone is missing is considered as a failure. I assume that instead of being it a bug in the code, the file format of the log file is not correctly formed.

@Nolski to solve your problem, you should add timezone information to your logs. For instance by adding "+0000" or whatever timezone you would like to use to your log file, as below. This should be easily be doable with a small script.

$ cat ./supybot/logs/ChannelLogger/freenode/#rit-foss/#rit-foss.log
2021-01-12T11:28:19+0000  *** grimoire_bot <grimoire_bot!~limnoria@45.55.49.14> has joined #rit-foss
2021-01-12T11:29:41+0000  <nolski> test
2021-01-12T11:29:41+0000  <xbot1313> Test Case Passed!

**Comment by Nolski** [comment]
@gregoriorobles Given that this is the log that supybot generates, it would be worthwhile to have perceval conform to that format right? I'm curious if other grimoire users are having issues with the supybot integration.

**Comment by gregoriorobles** [comment]
@Nolski I am not a core developer of perceval. If @sduenas or any other core developer of perceval are positive about what you propose, I could implement it.

**Comment by Nolski** [comment]
For what it's worth, these logs are generated by a bot running [Limnoria](https://github.com/ProgVal/Limnoria) which is the successor to Supybot as Supybot has been deprecated

**Comment by sduenas** [comment]
This was fixed by #716.
